examples/games/desktoy.rs

```diff
@@ -0,0 +1,389 @@
+#[derive(Component)]
+struct Pupil {
+    /// Radius of the eye containing the pupil.
+    eye_radius: f32,
+    /// Radius of the pupil.
+    pupil_radius: f32,
+    /// Current velocity of the pupil.
+    velocity: Vec2,
+}
+
+// Dimensions are based on: assets/branding/icon.png
+// Bevy logo radius
+const BEVY_LOGO_RADIUS: f32 = 128.0;
+// Birds' eyes x y (offset from the origin) and radius
+// These values are manually determined from the logo image
+const BIRDS_EYES: [(f32, f32, f32); 3] = [
+    (145.0 - 128.0, -(56.0 - 128.0), 12.0),
+    (198.0 - 128.0, -(87.0 - 128.0), 10.0),
+    (222.0 - 128.0, -(140.0 - 128.0), 8.0),
+];
+
+/// Spawn the scene
+fn setup(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    // Spawn a 2D camera
+    commands.spawn(Camera2dBundle::default());
+
+    // Spawn the text instructions
+    let font = asset_server.load("fonts/FiraSans-Bold.ttf");
+    let text_style = TextStyle {
+        font: font.clone(),
+        font_size: 30.0,
+        color: Color::WHITE,
+    };
+    commands.spawn((
+        Text2dBundle {
+            text: Text::from_section(
+                "Press Space to play on your desktop! Press it again to return.\nRight click Bevy logo to exit.",
+                text_style.clone(),
+            ),
+            transform: Transform::from_xyz(0.0, -300.0, 100.0),
+            ..default()
+        },
+        InstructionsText,
+    ));
+
+    // Create a circle mesh. We will reuse this mesh for all our circles.
+    let circle = Mesh2dHandle(meshes.add(Circle { radius: 1.0 }));
+    // Create the different materials we will use for each part of the eyes. For this demo they are basic [`ColorMaterial`]s.
+    let outline_material = materials.add(Color::BLACK);
+    let sclera_material = materials.add(Color::WHITE);
+    let pupil_material = materials.add(Color::srgb(0.2, 0.2, 0.2));
+    let pupil_highlight_material = materials.add(Color::srgba(1.0, 1.0, 1.0, 0.2));
+
+    // Spawn the Bevy logo sprite
+    commands
+        .spawn((
+            SpriteBundle {
+                texture: asset_server.load("branding/icon.png"),
+                ..default()
+            },
+            BevyLogo,
+        ))
+        .with_children(|commands| {
+            // For each bird eye
+            for (x, y, radius) in BIRDS_EYES {
+                // eye outline
+                commands.spawn(MaterialMesh2dBundle {
+                    mesh: circle.clone(),
+                    material: outline_material.clone(),
+                    transform: Transform::from_xyz(x, y - 1.0, 1.0).with_scale(Vec3::new(
+                        radius + 2.0,
+                        radius + 2.0,
+                        1.0,
+                    )),
+                    ..default()
+                });
+
+                // sclera
+                commands
```

Comment:
Are these two `SpatialBundle`s totally necessary?

Reply:
The first `SpatialBundle` is to avoid propagating the scale of the sclera to pupil. The second one is to move the pupil and its highlight as one unit.